### ISSUE TITLE
Fix respectful formatting of list comprehensions

### ISSUE THREAD

**Issue by vherrmann** [fourmolu/fourmolu#171]
**Is your feature request specific to Fourmolu?**
- [x] Yes, this feature is not relevant to Ormolu
 
Currently fourmolu (with default, respectful settings) formats
```hs
test = [ x | i <- [0..10],
          let x = i +1,
          x /= 0
  ]
```
to 
```hs
test =
    [ x | i <- [0 .. 10], let x = i + 1, x /= 0
    ]
```
I think in respectful mode (with leading commas), fourmolu should format it to sth like:
```hs
test = [ x | i <- [0..10]
               , let x = i +1
               , x /= 0 ]
```
Though I don't really care that much how it's indented, as long as fourmolu leaves the staments on seperate lines.

Related: #127

**Comment by georgefst** [comment]
> I think in respectful mode (with leading commas), fourmolu should format it to sth like:
> 
> ```haskell
> test = [ x | i <- [0..10]
>            , let x = i +1
>            , x /= 0 ]
> ```

Given that diff-friendliness is very important to us, I think the idiomatic output would be:
```hs
test =
    [ x | i <- [0..10]
        , let x = i +1
        , x /= 0
    ]
```

**Comment by philderbeast** [comment]
Is this an option here?

```haskell
test =
    [ x
    | i <- [0..10]
    , let x = i +1
    , x /= 0
    ]
```

**Comment by georgefst** [comment]
Yes, that also seems a reasonable output, consistent with Fourmolu's style.

Ideally I guess the decision of whether to put `|` on a separate line would be based on user input, and the indentation would be a boolean flag.
